### PR TITLE
Some more API v2 tweaks

### PR DESCRIPTION
**Pull request by rowanseymour**
* Added convenient mechanism for list endpoints to enforce mutually exclusive params
* Don't allow combining contact with other filters on messages endpoint
* Don't allow both contact and flow on runs endpoint
* Fixed format of error response when exception thrown from get_queryset
* Don't return deleted messages from messages endpoint
* Added missing id params on read explorers
* Moar tests

### CODE REVIEW

**Comment by rowanseymour** [comment]
@nicpottier been thinking and not allowing combining contact with other params for messages and runs seems exactly like the sort of thing that fails the _has anyone said they need this potentially expensive thing_ test... so have disabled that for now.

File: temba/api/v2/views.py [hunk 1]
```diff
@@ -320,28 +328,22 @@ class MessagesEndpoint(ListAPIMixin, BaseAPIView):
...
             sys_label = self.FOLDER_FILTERS.get(folder.lower())
             if sys_label:
-                return SystemLabel.get_queryset(self.request.user.get_org(), sys_label)
+                return SystemLabel.get_queryset(org, sys_label, exclude_test_contacts=False)
             else:
-                return Msg.current_messages.filter(pk=-1)
+                return self.model.current_messages.filter(pk=-1)
         else:
-            return super(MessagesEndpoint, self).get_queryset()
+            return self.model.current_messages.filter(org=org).exclude(visibility=DELETED).exclude(msg_type=None)
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: I'm still not 100% sure it's not possible to miss messages that haven't yet been handled...

File: temba/api/support.py [hunk 2]
```diff
@@ -91,6 +92,13 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
         return super(DocumentationRenderer, self).render(data, accepted_media_type, renderer_context)
 
 
+class InvalidQueryError(APIException):
```
Comment (rowanseymour) [inline temba/api/support.py]: `ValidationError` exceptions don't render correctly if you throw them from anything other than a serializer. There's no other built in exception type for 400 errors, but this custom one will render as `{"detail": ...}` same as other error types.

File: temba/api/v2/views.py (hunk 1 above)
Comment (nicpottier) [inline temba/api/v2/views.py]: We should definitely get to the bottom of that and make sure it is the case. Vital piece to making this all work.

Comment (rowanseymour) [inline temba/api/v2/views.py]: Will do but to be clear nothing in this PR effects that

**Pull request merged**